### ISSUE TITLE
Attention in CV

### ISSUE THREAD

**Issue by yufang67** [yufang67/machine-learning-articles#15]
# TL;DR
## Self-Attention

![self_attention_image](https://mmbiz.qpic.cn/sz_mmbiz_png/gYUsOT36vfr4TRS18Vww6JKBmT8l4icU62MoiarthxLibAcZdEfia0WKDotpoCiajoJGAqAxszYWUZ4icBMcPPtYDbdA/640?wx_fmt=png&tp=webp&wxfrom=5&wx_lazy=1&wx_co=1)

![self_attention_image](https://mmbiz.qpic.cn/sz_mmbiz_png/gYUsOT36vfr4TRS18Vww6JKBmT8l4icU6VHfhoAS3Chzu8QI6hRjsvH9tib3HLZVnGEcNRdqA7UZHOaVUE8V18Ww/640?wx_fmt=png&tp=webp&wxfrom=5&wx_lazy=1&wx_co=1)

![self_attention_image](https://mmbiz.qpic.cn/sz_mmbiz_png/gYUsOT36vfr4TRS18Vww6JKBmT8l4icU6wElcicGqgecxCZmbanzpBaXEXsUNM1V4PibuAomN5icbpU4ibcHSKaWptw/640?wx_fmt=png&tp=webp&wxfrom=5&wx_lazy=1&wx_co=1)

### Spatial transformer network
[paper](http://papers.nips.cc/paper/5854-spatial-transformer-networks)
[gitHub](https://github.com/fxia22/stn.pytorch)

### Channel Attention
[Squeeze-and-Excitation Networks](https://arxiv.org/abs/1709.01507)
- SEBlock
 将输入特征进行 Global avgpooling，得到1×1×Channel
 然后bottleneck特征交互一下，先压缩channel数，再重构回channel数
 最后接个sigmoid，生成channel间0~1的attention weights，最后scale乘回原输入特征

### Mix (spatial and channel)
Residual Attention Network for image classification
Dual Attention Network for Scene Segmentation

### Non-local
non-local neural networks
[gitHub](https://github.com/AlexHex7/Non-local_pytorch)

### Point-wise attention
CCNet: Criss-Cross Attention for Semantic Segmentation
[github](https://github.com/speedinghzl/CCNet)

### Hard attention

# Ref
[综述｜计算机视觉中的注意力机制](https://zhuanlan.zhihu.com/p/146130215)
